lib/views/movie_list_view.dart

```diff
@@ -0,0 +1,60 @@
+import 'package:flutter/material.dart';
+
+import '../custom_widgets/movie_info.dart';
+import '../repositories/movie_repository.dart';
+import '../entities/movie.dart';
+
+class MovieListView extends StatefulWidget {
+  const MovieListView({super.key});
+
+  @override
+  State<MovieListView> createState() => _MovieListViewState();
+}
+
+class _MovieListViewState extends State<MovieListView> {
+  final MovieRepository movieRepository = MovieRepository();
+  Future<List<Movie>>? movies;
+  static const double dividerHeight = 20;
+  static const String errorMessage = 'An error has occurred while loading data';
+
+  @override
+  void initState() {
+    super.initState();
+    movies = movieRepository.getMovies();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return FutureBuilder(
+      future: movies,
+      builder: (
+        BuildContext context,
+        AsyncSnapshot<List<Movie>> snapshot,
+      ) {
+        if (snapshot.connectionState == ConnectionState.waiting) {
+          return const Center(
+            child: CircularProgressIndicator(
+              color: Colors.white,
+            ),
+          );
+        } else if (snapshot.hasData) {
+          List<Movie>? movies = snapshot.data;
+          return ListView.separated(
+            itemCount: movies!.length,
```

Comment:
I wouldn't force a null-unwrapping here. I mean, we all know that if `snapshot.hasData`, `snapshot.data` won't be null. 
But just for the sake of using Dart's null safety features, I'd make this change:
`List<Movie> movies = snapshot.data ?? [];`

Reply:
jajaj, fran made a comment related to this above, and suggested something similar but not exactly the same... so, to make you both happy, i implemented this solution with movies snapshot data, and did fran solution to genres snapshot data jaaja.... hope you don't mind!! jaajaj